AzureTable/Microsoft.DataTransfer.AzureTable/Source/AzureTableSourceAdapter.cs

```diff
@@ -53,35 +59,53 @@ public AzureTableSourceAdapter(IAzureTableSourceAdapterInstanceConfiguration con
 
         public async Task<IDataItem> ReadNextAsync(ReadOutputByRef readOutput, CancellationToken cancellation)
         {
-            if (segmentDownloadTask == null)
-            {
-                MoveToNextSegment(null, cancellation);
-            }
+            TableQuerySegment<DynamicTableEntity> currentSegment = null;
+            RemoteLoggingClientProvider remoteLoggingClientProvider = new RemoteLoggingClientProvider();
+            IRemoteLogging remoteLogger = remoteLoggingClientProvider.GetRemoteLogger("tableapibulk");
 
-            var currentSegment = await segmentDownloadTask;
-
-            // Make sure current segment has data to read
-            while (currentEntityIndex >= currentSegment.Results.Count && currentSegment.ContinuationToken != null)
+            try
             {
-                MoveToNextSegment(currentSegment.ContinuationToken, cancellation);
+                if (segmentDownloadTask == null)
+                {
+                    MoveToNextSegment(null, cancellation);
+                }
+
                 currentSegment = await segmentDownloadTask;
-            }
 
-            if (currentEntityIndex >= currentSegment.Results.Count && currentSegment.ContinuationToken == null)
-            {
-                return null;
-            }
+                // Make sure current segment has data to read
+                while (currentEntityIndex >= currentSegment.Results.Count && currentSegment.ContinuationToken != null)
+                {
+                    MoveToNextSegment(currentSegment.ContinuationToken, cancellation);
+                    currentSegment = await segmentDownloadTask;
+                }
+
+                if (currentEntityIndex >= currentSegment.Results.Count && currentSegment.ContinuationToken == null)
+                {
+                    return null;
+                }
 
-            var entity = currentSegment.Results[currentEntityIndex++];
-            readOutput.DataItemId = entity.RowKey;
+                var entity = currentSegment.Results[currentEntityIndex++];
+                readOutput.DataItemId = entity.RowKey;
 
-            if (currentEntityIndex >= currentSegment.Results.Count && currentSegment.ContinuationToken != null)
+                if (currentEntityIndex >= currentSegment.Results.Count && currentSegment.ContinuationToken != null)
+                {
+                    // Start downloading next segment while current record is being processed
+                    MoveToNextSegment(currentSegment.ContinuationToken, cancellation);
+                }
+
+                return new DynamicTableEntityDataItem(AppendInternalProperties(entity));
+            }
+            catch(Exception excp)
```

Comment:
Use standard .NET formatting guide lines please, add space between catch and (

Reply:
N/A anymore.